flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/NFA.java

```diff
@@ -157,15 +157,28 @@
 	 */
 	private boolean nfaChanged;
 
+	/**
+	 * Store the skip strategy.
+	 */
+	private AfterMatchSkipStrategy afterMatchSkipStrategy;
```

Comment:
I would not add it to NFA. I think it is a static data, that should not be serialized each time with `NFA`. We already have problems to remove `State`s, `Condition`s and `handleTimeout` from serializing within `NFA`. It also makes maintaining serializing compatbility hard. How about adding it to the `NFAFactory` and passing it to `NFA#process`(It is the only place where it is needed).

I know it will require passing it in multiple places e.g. `KeyedCEPPatternOperator` and  'TimeoutKeyedCEPPatternOperator` but it will be easier after #4320.